src/redux/features/taskSlice.ts

```diff
@@ -25,11 +26,60 @@ export const taskSlice = createSlice({
         state.tasks[index] = { ...action.payload };
       }
     },
+    handleCompleteSubtask: (
+      state,
+      action: PayloadAction<{ taskId: string; subtaskId: string }>
+    ) => {
+      const { taskId, subtaskId } = action.payload;
+      const taskIndex = state.tasks.findIndex((task) => task.id === taskId);
+      if (taskIndex !== -1) {
+        const subtaskIndex = state.tasks[taskIndex].subtasks.findIndex(
```

Comment:
You can take `state.tasks[taskIndex]` into a variable as it is being used at multiple places.